DEVELOPER_GUIDE.md

```diff
@@ -202,6 +203,15 @@ You can now import the OpenSearch project into Eclipse as follows.
 
 **Note:** it may look non-intuitive why one needs to use Gradle wrapper and then import existing Gradle project (in general, **File > Import -> Existing Gradle Project** should be enough). Practically, as it stands now, Eclipse Buildship plugin does not import OpenSearch project dependencies correctly but does work in conjunction with Gradle wrapper.
 
+### JetBrains Gateway
+
+[JetBrains Gateway](https://www.jetbrains.com/remote-development/gateway/) enables development, testing and debugging on remote machines like development servers.
```

Comment:
Just curious, this seems to be nothing specific to opensearch. Do we still want to have this in the docs?

Reply:
This is something which helped me as a contributor to get started quickly with remote debugging and development, given the nature of the server requirements for OpenSearch.

Reply:
I think the confusing part is that JetBrains Gateway is not an editor. How about moving this section under "IntelliJ IDEA"?

VSCode has a similar remote host setup out of the box with no additional work, so we don't mention it.

Reply:
Moved the section under remote development for IntelliJ.